sdk/core/azure-core/src/main/java/com/azure/core/util/logging/ClientLogger.java

```diff
@@ -237,37 +294,44 @@ private void performLogging(LogLevel logLevel, LogLevel environmentLogLevel, boo
         }
     }
 
-    /*
-     * Determines if the environment and logger support logging at the given log level.
+    /**
+     * Determines if the app or environment logger support logging at the given log level.
      *
      * @param logLevel Logging level for the log message.
-     * @param environmentLoggingLevel Logging level the environment is set to support.
      * @return Flag indicating if the environment and logger are configured to support logging at the given log level.
      */
-    private boolean canLogAtLevel(LogLevel logLevel, LogLevel environmentLoggingLevel) {
-        // Do not log if logLevel is null is not set.
-        if (logLevel == null) {
-            return false;
-        }
+    public boolean canLogAtLevel(LogLevel logLevel) {
+        LogLevel rootLogLevel = loadSystemLogLevel();
+        return canLogAtLevel(logLevel, rootLogLevel);
+    }
 
-        // Attempting to log at a level not supported by the environment.
-        if (logLevel.getLogLevel() < environmentLoggingLevel.getLogLevel()) {
-            return false;
-        }
+    private boolean canLogAtLevel(LogLevel logLevel, LogLevel allowedLogLevel) {
+        // Attempting to log at a level not supported by the SLF4J configuration or env variable.
+        return logLevel.getLogLevel() >= allowedLogLevel.getLogLevel();
+    }
 
-        // Determine if the logger configuration supports logging at the level.
-        switch (logLevel) {
-            case VERBOSE:
-                return logger.isDebugEnabled();
-            case INFORMATIONAL:
-                return logger.isInfoEnabled();
-            case WARNING:
-                return logger.isWarnEnabled();
-            case ERROR:
-                return logger.isErrorEnabled();
-            default:
-                return false;
+    /**
+     * Checking the system log with the preference order of slf4j and environment variable.
+     *
+     * @return The log level.
+     */
+    private LogLevel loadSystemLogLevel() {
```

Comment:
This method name is not clear enough. You aren't 'loading', this is more like a 'getConfiguredLogLevel()' or something?

Reply:
Done.